base_agent/dialogue_objects/get_memory_handler.py

```diff
@@ -168,9 +168,10 @@ def do_answer(self, mems: Sequence[Any], vals: Sequence[Any]) -> Tuple[Optional[
                 return self.handle_exists(mems)
             else:
                 raise ValueError("Bad answer_type={}".format(output_type))
+        except IndexError: # index error indicates no answer available
```

Comment:
Can you also add a logging statement for the `IndexError` here? We shouldn't be catching errors silently. Adding a logging statement here will help disambiguate the `ErrorWithResponse` thrown here from all the other places that `ErrorWithResponse` could arise from.